test/extension/handlers/lifecycle/handlers.go

```diff
@@ -36,121 +41,161 @@ import (
 	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
 )
 
-// Handler is the handler for the lifecycle hooks.
-type Handler struct {
-	Client client.Client
+// Manager is the Manager for the lifecycle hooks handlers;
+// it provides a common struct shared across all the lifecycle hook handlers because all of them
+// are using a controller runtime client and the same set of func to work with the config map where
+// preloaded answers for lifecycle hooks are stored.
+// NOTE: it is not mandatory to use a Manager in custom RuntimeExtension, what is important
+// is to expose HandlerFunc with the signature defined in sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1.
+type Manager struct {
+	client client.Client
 }
 
-// DoBeforeClusterCreate implements the BeforeClusterCreate hook.
-func (h *Handler) DoBeforeClusterCreate(ctx context.Context, request *runtimehooksv1.BeforeClusterCreateRequest, response *runtimehooksv1.BeforeClusterCreateResponse) {
+// NewManager returns a Manager for the lifecycle hooks handlers.
+func NewManager(scheme *runtime.Scheme) (*Manager, error) {
+	restConfig, err := ctrl.GetConfig()
```

Comment:
Should we really create our own client inside of this manager? This means we can't share caching etc across the entire RuntimeExtension (compare with creating our own client inside of a webhook or a controller in a regular capi controller)

Reply:
Fixed